functional-tests/src/test/java/org/infinispan/online/service/endpoint/RESTTester.java

```diff
@@ -37,113 +49,156 @@ public void testPutPerformance(URL urlToService, long timeout, TimeUnit timeUnit
-      post(urlServerAddress, key, value, 200);
+   private void post(String url, String body, int expectedCode, boolean authenticate) {
+      HttpClient client;
+      try {
+         client = httpClient(url, authenticate);
+         Request req = client.POST(url);
+         req.header(HttpHeader.ACCEPT, "text/plain");
+         req.header(HttpHeader.CONTENT_TYPE, "text/plain");
+         req.content(new StringContentProvider(body));
+         req.timeout(60, TimeUnit.SECONDS);
+         sendAndAssertRequest(req, expectedCode);
+      } catch (Exception e) {
+         throw new IllegalStateException(e);
+      }
    }
 
-   public void put(String urlServerAddress, String key, String value, int expectedCode) {
-      put(urlServerAddress.toString() + "rest/" + cache + "/" + key, value, expectedCode, true);
+   private void put(String url, String body) {
+      HttpClient client;
+      try {
+         client = httpClient(url);
+         Request req = client.newRequest(url).method(HttpMethod.PUT);
+         req.header(HttpHeader.ACCEPT, "text/plain");
+         req.header(HttpHeader.CONTENT_TYPE, "text/plain");
+         req.content(new StringContentProvider(body));
+         sendAndAssertRequest(req, 200);
+      } catch (Exception e) {
+         throw new IllegalStateException(e);
+      }
    }
 
-   public void put(String urlServerAddress, String key, String value) {
-      put(urlServerAddress, key, value, 200);
+   private HttpClient httpClient(String url) throws Exception {
+      return httpClient(url, true);
+   }
+
+   private HttpClient httpClient(String url, boolean authenticate) throws Exception {
+      HttpClient httpClient;
+      SslContextFactory sslContextFactory = new SslContextFactory();
+      sslContextFactory.setTrustStorePath(trustStorePath);
+      sslContextFactory.setTrustStorePassword(new String(TrustStore.TRUSTSTORE_PASSWORD));
+      sslContextFactory.setKeyStorePath(trustStorePath);
+      sslContextFactory.setKeyStorePassword(new String(TrustStore.TRUSTSTORE_PASSWORD));
+      httpClient = new HttpClient(sslContextFactory);
+      httpClient.setConnectTimeout(TimeUnit.SECONDS.toMillis(60));
+      httpClient.start();
+
+      URI uri = URI.create(url);
+      AuthenticationStore authStore = httpClient.getAuthenticationStore();
+      if (authenticate) {
+         authStore.addAuthenticationResult(new BasicAuthentication.BasicResult(uri, "test", "test"));
+      } else {
+         Authentication.Result authResult = authStore.findAuthenticationResult(uri);
+         if (authResult != null)
+            authStore.removeAuthenticationResult(authResult);
+      }
+      return httpClient;
    }
 
-   public String getMethod(String urlServerAddress, String key, int expectedCode) {
-      return get(urlServerAddress.toString() + "rest/" + cache + "/" + key, expectedCode, true);
+   private ContentResponse sendAndAssertRequest(Request req, int expectedCode) throws Exception {
```

Comment:
TBH, I'm not a fan of sending request and validating the resulting code in a single method. Would it be possible to refactor the tests to return the ContentResponse and then, inside a test, validate if returned code was fine or not? We could use AssertJ's custom assertions for that.